x-pack/plugins/security_solution/common/detection_engine/build_exceptions_filter.ts

```diff
@@ -0,0 +1,303 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Filter } from '../../../../../src/plugins/data/common';
+import {
+  ExceptionListItemSchema,
+  CreateExceptionListItemSchema,
+  EntryMatch,
+  EntryMatchAny,
+  EntryNested,
+  entriesMatch,
+  entriesMatchAny,
+  entriesExists,
+  entriesNested,
+  EntryExists,
+} from '../../../lists/common';
+import { BooleanFilter, NestedFilter } from './types';
+import { hasLargeValueList } from './utils';
+
+type NonListEntry = EntryMatch | EntryMatchAny | EntryNested | EntryExists;
+interface ExceptionListItemNonLargeList extends ExceptionListItemSchema {
+  entries: NonListEntry[];
+}
+
+interface CreateExceptionListItemNonLargeList extends CreateExceptionListItemSchema {
+  entries: NonListEntry[];
+}
+
+export type ExceptionItemSansLargeValueLIsts =
+  | ExceptionListItemNonLargeList
+  | CreateExceptionListItemNonLargeList;
+
+export const chunkExceptions = (
+  exceptions: ExceptionItemSansLargeValueLIsts[],
+  chunkSize: number
+): ExceptionItemSansLargeValueLIsts[][] => {
+  if (exceptions.length === 0) {
+    return [];
+  } else if (exceptions.length <= chunkSize) {
+    return [exceptions];
+  } else {
+    const chunkedFilters: ExceptionItemSansLargeValueLIsts[][] = [];
+    for (let index = 0; index < exceptions.length; index += chunkSize) {
+      const exceptionsChunks = exceptions.slice(index, index + chunkSize);
+      chunkedFilters.push(exceptionsChunks);
+    }
+    return chunkedFilters;
+  }
+};
```

Comment:
I've caught myself writing my own chunking from time to time even though I always forget about lodash having one:

```ts
import { chunk } from 'lodash/fp';

export const chunkExceptions = (
  exceptions: ExceptionItemSansLargeValueLIsts[],
  chunkSize: number
): ExceptionItemSansLargeValueLIsts[][] => {
  return chunk(chunkSize, exceptions);
};
```

That looks like it will pass your tests fwiw if you want to just use lodash:
https://lodash.com/docs/4.17.15#chunk

Reply:
Updated, thanks!